src/GovUk.Education.ExploreEducationStatistics.Data.Importer/Services/ImporterLocationService.cs

```diff
@@ -166,20 +166,13 @@ public class ImporterLocationService : BaseImporterService
                 .And(location => location.Institution.Code == (institution != null ? institution.Code : null) 
                                  && location.Institution.Name == (institution != null ? institution.Name : null));
 
-            // If no LA code then try using the old code
-            if (localAuthority != null && localAuthority.Code == null)
-            {
-                predicateBuilder = predicateBuilder
-                    .And(location => location.LocalAuthority.OldCode == (localAuthority != null && localAuthority.OldCode != null ? localAuthority.OldCode : null)
-                                     && location.LocalAuthority.Name == (localAuthority != null ? localAuthority.Name : null));
-            }
-            else
-            {
-                predicateBuilder = predicateBuilder
-                    .And(location => location.LocalAuthority.Code == (localAuthority != null && localAuthority.Code != null ? localAuthority.Code : null)
-                                     && location.LocalAuthority.Name == (localAuthority != null ? localAuthority.Name : null)); 
-            }
-        
+            // Also match the old LA code even if blank
+            predicateBuilder = predicateBuilder
+                .And(location =>
+                    location.LocalAuthority.Code == (localAuthority != null && localAuthority.Code != null ? localAuthority.Code : null)
```

Comment:
I'm not mistaken, we could use the elvis operator for these null checks

Reply:
You can't use the null-conditional operator `x?.y` in expression lamdas here because it's not supported by the LINQ provider. If you try it you will get the compile error `An expression tree lamda may not contain conditional access expression`

Reply:
Cool, didn't know that!